### PR TITLE
feat(ci): map pool projects kube-agents-evals-16 through evals-30 to GitOps repos

### PR DESCRIPTION
**Pull request by lapis2002**
## Summary

Maps evaluation pool projects `kube-agents-evals-16` through `kube-agents-evals-30` to their dedicated private GitOps infrastructure repositories (`gke-agentic/kube-agents-evals-16-infra` through `evals-30-infra`) across `hack/ci-deploy.sh`, `tests/test_ci_gitops_repo.py`, and the CI pool deployment documentation.

## Why This Change

Doubles the Boskos evaluation pool capacity from 15 to 30 projects to accommodate increasing presubmit volume and eliminate Boskos lease queue wait times.

Precondition Step 0 of `scripts/provision_ci_pool_project.sh` asserts that the project is registered in `gitops_repo_for_project()` before applying Terraform/OpenTofu resources, preventing runs from leasing unmapped projects and failing with unmapped-project refusals.

## What Changed

- **`hack/ci-deploy.sh`**: Added case arms for `kube-agents-evals-16` through `30` returning `gke-agentic/kube-agents-evals-<N>-infra`.
- **`tests/test_ci_gitops_repo.py`**: Added `kube-agents-evals-16` through `30` to `_EXPECTED_MAPPING`.
- **`docs/site/src/content/docs/deploy/ci-pool-projects.md`**: Added rows for `evals-16` through `evals-30` to the GitOps repository catalog table.

## Context

- Scales the Boskos `kube-agents-evals-project` pool size from 15 to 30 projects.
- Pairs with the corresponding updates:
  - `gke-internal-test-infra`: Adding `kube-agents-evals-16..30` to Boskos pool overlays (`deployments/gke-agentic-tooling-team/boskos/overlays/boskos-resources.yaml`, Gerrit cl/2245750).
  - `oss-test-infra`: Bumping `max_concurrency: 15` → `max_concurrency: 30` in `pull-kube-agents-smoke-test` (GoogleCloudPlatform/oss-test-infra#2685).

## Testing

- `python3 -m unittest tests/test_ci_gitops_repo.py`: 15 passed (`OK`).
- `PYTHONPATH=scripts python3 -m unittest scripts/test_verify_ci_pool_project.py`: 218 passed (`OK`).
- `bash -n hack/ci-deploy.sh`: Clean syntax check.

### Live Validation

Not live-tested on a merged main runner: this change lands the static project mapping required before newly provisioned projects can be leased.

Verified locally against the verifier and deployment resolver:
```bash
$ python3 scripts/verify_ci_pool_project.py --project-id kube-agents-evals-16
[?] Codebase GitOps Mapping: Mapped to gke-agentic/kube-agents-evals-16-infra in this checkout, not yet on upstream/main
```
Confirmed that `verify_ci_pool_project.py` reads `gitops_repo_for_project()` from `hack/ci-deploy.sh` and correctly resolves `kube-agents-evals-16` to `gke-agentic/kube-agents-evals-16-infra`. Projects 18–30 are actively in the provisioning sequence.

## Self-Review

- **Context**: Both passes ran in the primary agent development session with direct file inspection and automated test suites.
- **Adversarial pass**: Inspected `hack/ci-deploy.sh` case arm formatting to ensure regex matching in `provision_ci_pool_project.sh:184` (`^[[:space:]]*${PROJECT_ID}\)[[:space:]]+echo[[:space:]]+"${GITOPS_REPO}"`) matches without syntax ambiguities. Checked boundary conditions for `_NEVER_MAPPED_PROJECT` (`not-a-pool-project-fixture`) ensuring fail-closed behavior on unmapped IDs. Verified 15 tests in `test_ci_gitops_repo.py` and 218 tests in `test_verify_ci_pool_project.py` pass without errors.
- **Docs drift pass**: Hand-verified `docs/site/src/content/docs/deploy/ci-pool-projects.md` table lines 177–195 to ensure sequential ordering (evals-16..30), correct Markdown formatting, and exact 1:1 parity with the case arms in `hack/ci-deploy.sh`. Ran `scripts/generate_docs.py --check` confirming no drift across mechanical targets.

## Risk & Rollout

Low risk. Does not modify runtime cluster agent logic, controller operators, or portal workloads. Revert requires no state migration.

### CODE REVIEW

File: docs/site/src/content/docs/deploy/ci-pool-projects.md [hunk 1]
```diff
@@ -177,6 +177,21 @@ The evaluation scenarios that exercise the GitOps workflow — the six fleet-aud
 | `kube-agents-evals-13` | `gke-agentic/kube-agents-evals-13-infra` |
```
[the remark this answers is not in the document]
Comment (lapis2002) [reply docs/site/src/content/docs/deploy/ci-pool-projects.md:177]: Updated

**Review by bradhoekstra — APPROVED** [review]

**Review by jayantid — APPROVED** [review]
